clippy_lints/src/methods.rs

```diff
@@ -913,6 +913,10 @@ fn lint_or_fun_call(cx: &LateContext, expr: &hir::Expr, name: &str, args: &[hir:
             return;
         }
 
+        let start_point = self_expr.span.hi();
+        let end_point = span.hi();
+        let span_replace_word = Span::new(start_point, end_point, span.ctxt());
```

Comment:
You can use `self_expr.span.end_point().to(span.hi())` I think. check the methods on `syntax::code_map::Span`

Reply:
Frustratingly thats the wrong Span object. It uses this one: http://manishearth.github.io/rust-internals-docs/syntax_pos/struct.Span.html - took me a while to realise.

However I have reduced it down to: 
`self_expr.span.with_lo(span.hi());`